Library/ViewModels/DiscoveryNavigationHeaderViewModel.swift

```diff
@@ -172,17 +171,14 @@ public final class DiscoveryNavigationHeaderViewModel: DiscoveryNavigationHeader
     self.showDiscoveryFilters = rowForFilters
       .takeWhen(paramsAndFiltersAreHidden.filter { !$0.filtersAreHidden })
 
-    self.subviewColor = primaryColor
+    self.subviewColor = categoryId.mapConst(discoveryPrimaryColor())
```

Comment:
Could `mapConst` on `viewDidLoad` here to be explicit about it not depending on `categoryId` perhaps.